modules/nw-metallb-software-components.adoc

```diff
@@ -5,31 +5,27 @@
 [id="nw-metallb-software-components_{context}"]
 = MetalLB software components
 
-When you install the MetalLB Operator, the `metallb-operator-controller-manager` deployment starts a pod.
-The pod is the implementation of the Operator.
-The pod monitors for changes to all the relevant resources.
+When you install the MetalLB Operator, the `metallb-operator-controller-manager` deployment starts a pod. The pod is the implementation of the Operator. The pod monitors for changes to all the relevant resources.
 
 When the Operator starts an instance of MetalLB, it starts a `controller` deployment and a `speaker` daemon set.
 
 `controller`::
-The Operator starts the deployment and a single pod.
-When you add a service of type `LoadBalancer`, Kubernetes uses the `controller` to allocate an IP address from an address pool.
+The Operator starts the deployment and a single pod. When you add a service of type `LoadBalancer`, Kubernetes uses the `controller` to allocate an IP address from an address pool.
+In case of a service failure, ensure you have the following entry in your `controller` pod logs:
++
+[source,terminal]
+----
+"event":"ipAllocated","ip":"172.22.0.201","msg":"IP address assigned by controller
+----
 
 `speaker`::
-The Operator starts a daemon set for `speaker` pods.
-By default, a pod is started on each node in your cluster.
-You can limit the pods to specific nodes by specifying a node selector in the `MetalLB` custom resource when you start MetalLB.
+The Operator starts a daemon set for `speaker` pods. By default, a pod is started on each node in your cluster. You can limit the pods to specific nodes by specifying a node selector in the `MetalLB` custom resource when you start MetalLB. If the `controller` allocated the ip address to the service and service is still unavailable, read the `speaker` pod logs. If the speaker pod is unavailable, run the `oc describe pod -n` command.
 +
 For layer 2 mode, after the `controller` allocates an IP address for the service, the `speaker` pods use an algorithm to determine which `speaker` pod on which node will announce the load balancer IP address.
-The algorithm involves hashing the node name and the load balancer IP address.
-See the section about external traffic policy for more information.
+The algorithm involves hashing the node name and the load balancer IP address. See the section about external traffic policy for more information.
```

Comment:
Should the name of the section also be added for clarification in the second sentence? Up to you though! 😀

Reply:
Addressed

Reply:
Added the section name with the link

Reply:
@xenolinux Hi my apologies, I should've been more clear! The link won't work becuase we don't put xrefs in a module. I mean't adding the name of the section, in this case like, 
> See "MetalLB` and external traffic policy" for more information.
 instead of describing it

Reply:
Addressed